src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= (table.length) * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHashCode(key);
+        Node<K, V> newNode = new Node<>(key, value);
 
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if ((key == null && currentNode.key == null)
+                        || (key != null && key.equals(currentNode.key))) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getHashCode(key);
+        Node<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+            if ((key == null && currentNode.key == null)
+                    || (key != null && key.equals(currentNode.key))) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHashCode(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        int newCapacity = DEFAULT_CAPACITY << 1;
+        Node<K, V>[] newTable = new Node[newCapacity];
```

Comment:
You should use the diamond operator with the type parameters to avoid unchecked assignment warnings. Change `new Node[newCapacity]` to `new Node<K, V>[newCapacity]`.

Reply:
but its not alloved to create generic array